src/lib.rs

```diff
@@ -375,30 +321,42 @@ impl<T: Send + UnwindSafe> UnwindSafe for ThreadLocal<T> {}
 
 struct RawIter<T: Send> {
     remaining: usize,
+    buckets: [*const UnsafeCell<Option<T>>; BUCKETS],
```

Comment:
I missed this during my initial review: I would strongly prefer if the iterator didn't make a whole copy of the buckets and instead just accessed the bucket array by reference.

Reply:
@Amanieu That would be a bit difficult to implement because the location of the buckets may be moved in memory in `IntoIter`, so we can't hold a pointer to it. So we would have to either box the `ThreadLocal` in `IntoIter` or remove the `RawIterMut` abstraction entirely.

Reply:
`RawIter` doesn't actually need to implement `Iterator`. It can use inherent methods instead where the pointer to the array of buckets is passed as a parameter.